test/functional/lua/command_line_completion_spec.lua

```diff
@@ -0,0 +1,167 @@
+local helpers = require('test.functional.helpers')(after_each)
+
+local clear = helpers.clear
+local eq = helpers.eq
+local funcs = helpers.funcs
+local exec_lua = helpers.exec_lua
+
+local get_completions = function(input, env)
+  return exec_lua("return vim._expand_pat(...)", '^' .. input, env)
```

Comment:
is the second retval from `vim._expand_pat` the prefix len?  then we should change this to `return {vim._expand_pat(...)}` and update the tests.

Reply:
I think the prefix len is not used anywhere by the tests, just the actual completion. Since Lua only keeps the first return value in this case, we are probably free to keep the tests as is?

Reply:
I mean we probably _want_ to test the prefix len as it is essential to completion working as expected. using `{}` will capture all return values as a numeric table.